src/nlr/NetworkLevelReasoner.cpp

```diff
@@ -77,17 +82,22 @@ void NetworkLevelReasoner::setBias( unsigned layer, unsigned neuron, double bias
 
 void NetworkLevelReasoner::addActivationSource( unsigned sourceLayer,
                                                 unsigned sourceNeuron,
-                                                unsigned targetLeyer,
+                                                unsigned targetLayer,
                                                 unsigned targetNeuron )
 {
-    _layerIndexToLayer[targetLeyer]->addActivationSource( sourceLayer, sourceNeuron, targetNeuron );
+    _layerIndexToLayer[targetLayer]->addActivationSource( sourceLayer, sourceNeuron, targetNeuron );
 }
 
 const Layer *NetworkLevelReasoner::getLayer( unsigned index ) const
 {
     return _layerIndexToLayer[index];
 }
 
+Layer *NetworkLevelReasoner::getLayerToUpdate( unsigned index )
```

Comment:
Maybe just call this function "getLayer"?

Reply:
It causes problems because there is a **const** getLayer(). Or do you mean to overload with a non-const version?